File: ios/LaneShadow/Sandbox/Stories/Organisms/LSMapLayerStory.swift
import LaneShadowTheme
import NativeSandbox
import SwiftUI

@MainActor
enum LSMapLayerStory {
    static let all: [Story] = [
        // Story 1: Map Only
        Story(
            id: "organisms.maplayer.mapOnly",
            tier: .organism,
            component: "MapLayer",
            name: "Map Only",
            summary: "LSMap(mode:.preview) fills the entire screen. No other slots populated."
        ) { _ in
            LSMapLayer {
                LSMap(
                    mode: .preview,
                    camera: CameraPosition(
                        center: LatLng(lat: 37.7749, lon: -122.4194),
                        zoom: 12.0
                    )
                )
            }
        },

        // Story 2: Map + TopBar
        Story(
            id: "organisms.maplayer.mapTopBar",
            tier: .organism,
            component: "MapLayer",
            name: "Map + TopBar",
            summary: "LSTopBar at z-index 5, above everything. Map below. Matches Idle screen pattern."
        ) { _ in
            LSMapLayer(
                map: {
                    LSMap(
                        mode: .preview,
                        camera: CameraPosition(
                            center: LatLng(lat: 37.7749, lon: -122.4194),
                            zoom: 12.0
                        )
                    )
                },
                topBar: LSTopBar(
                    onMenuTap: {},
                    onNewTap: {}
                )
            )
        },

        // Story 3: Map + Top Overlay
        Story(
            id: "organisms.maplayer.mapTopOverlay",
            tier: .organism,
            component: "MapLayer",
            name: "Map + Top Overlay",
            summary: "NavigatorMessage in topOverlays slot. Positioned below TopBar with safe-area padding."
        ) { _ in
            LSMapLayer(
                map: {
                    LSMap(
                        mode: .preview,
                        camera: CameraPosition(
                            center: LatLng(lat: 37.7749, lon: -122.4194),
                            zoom: 12.0
                        )
                    )
                },
                topOverlays: [
                    GlassOverlaySlot(id: "navigator") {
                        LSGlassPanel(variant: .chrome) {
                            VStack(alignment: .leading, spacing: 8) {
                                LSText("THE NAVIGATOR", variant: .label.sm, color: .primary)
                                LSText(
                                    "Here are three routes south through the coastal hills.",
                                    variant: .heading.md
                                )
                            }
                            .frame(maxWidth: .infinity, alignment: .leading)
                        }
                    }
                ],
                topBar: LSTopBar(
                    onMenuTap: {},
                    onNewTap: {}
                )
            )
        },

        // Story 4: Map + Bottom Overlay
        Story(
            id: "organisms.maplayer.mapBottomOverlay",
            tier: .organism,
            component: "MapLayer",
            name: "Map + Bottom Overlay",
            summary: "ChatInput in bottomOverlays. Anchored above bottom safe area at z-index 2."
        ) { _ in
            LSMapLayer(
                map: {
                    LSMap(
                        mode: .preview,
                        camera: CameraPosition(
                            center: LatLng(lat: 37.7749, lon: -122.4194),
                            zoom: 12.0
                        )
                    )
                },
                bottomOverlays: [
                    GlassOverlaySlot(id: "chat") {
                        LSGlassPanel(variant: .chrome) {
                            HStack(spacing: 8) {
                                LSText(
                                    "Where do you want to ride?",
                                    variant: .body.md
                                )
                                Spacer()
                                LSIcon(name: .send, size: .sm, color: .primary)
                            }
                        }
                        .frame(height: 46)
                    }
                ],
                topBar: LSTopBar(
                    onMenuTap: {},
                    onNewTap: {}
                )
            )
        },

        // Story 5: Map + Scrim + Drawer
        Story(
            id: "organisms.maplayer.mapScrimDrawer",
            tier: .organism,
            component: "MapLayer",
            name: "Map + Scrim + Drawer",
            summary: "LSScrim z-1, SessionsDrawer at z-4 above scrim. Matches SCR-05 Sessions pattern."
        ) { _ in
            LSMapLayer(
                map: {
                    LSMap(
                        mode: .preview,
                        camera: CameraPosition(
                            center: LatLng(lat: 37.7749, lon: -122.4194),
                            zoom: 12.0
                        )
                    )
                },
                scrim: ScrimSpec(opacity: 0.35),
                leadingDrawer: DrawerSpec {
                    LSGlassPanel(variant: .chrome) {
                        VStack(alignment: .leading, spacing: 12) {
                            LSText("Rides", variant: .heading.md)
                            LSText("This Week", variant: .label.sm)
                            Rectangle()
                                .fill(Color(uiColor: .separator))
                                .frame(height: 1)
                            LSText("Santa Cruz Loop", variant: .body.md)
                            LSText("Skyline to the Sea", variant: .body.md)
                            LSText("PCH Evening Run", variant: .body.md)
                        }
                        .frame(maxWidth: .infinity, alignment: .leading)
                    }
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
                } onDismiss: {
                    // Dismiss handler
                }
            )
        },

        // Story 6: Map + Sheet
        Story(
            id: "organisms.maplayer.mapSheet",
            tier: .organism,
            component: "MapLayer",
            name: "Map + Sheet",
            summary: "RouteSheet at z-index 3, anchored bottom. TopBar still visible above sheet at z-5."
        ) { _ in
            LSMapLayer(
                map: {
                    LSMap(
                        mode: .preview,
                        camera: CameraPosition(
                            center: LatLng(lat: 37.7749, lon: -122.4194),
                            zoom: 12.0
                        )
                    )
                },
                bottomSheet: BottomSheetSpec(detent: .medium) {
                    VStack(alignment: .leading, spacing: 12) {
                        LSText("The Skyline Spine", variant: .heading.lg)
                        LSText("via Kings Mountain Rd · 47 mi", variant: .body.sm)
                        HStack(spacing: 8) {
                            LSGlassPanel(variant: .chrome) {
                                LSText("Save", variant: .label.md)
                            }
                            LSGlassPanel(variant: .chrome) {
                                LSText("Ride this", variant: .label.md, color: .primary)
                            }
                        }
                    }
                    .frame(maxWidth: .infinity, alignment: .leading)
                },
                topBar: LSTopBar(
                    onMenuTap: {}
                )
            )
        },

        // Story 7: Full Stack
        Story(
            id: "organisms.maplayer.fullStack",
            tier: .organism,
            component: "MapLayer",
            name: "Full Stack",
            summary: "All slots populated: map + scrim + top/bottom overlays + sheet + top bar. Maximum composition."
        ) { _ in
            LSMapLayer(
                map: {
                    LSMap(
                        mode: .preview,
                        camera: CameraPosition(
                            center: LatLng(lat: 37.7749, lon: -122.4194),
                            zoom: 12.0
                        )
                    )
                },
                scrim: ScrimSpec(opacity: 0.2),
                topOverlays: [
                    GlassOverlaySlot(id: "navigator") {
                        LSGlassPanel(variant: .chrome) {
                            VStack(alignment: .leading, spacing: 4) {
                                LSText("THE NAVIGATOR", variant: .label.sm, color: .primary)
                                LSText("Best route selected.", variant: .heading.sm)
                            }
                            .frame(maxWidth: .infinity, alignment: .leading)
                        }
                    }
                ],
                bottomOverlays: [
                    GlassOverlaySlot(id: "chat") {
                        LSGlassPanel(variant: .chrome) {
                            HStack(spacing: 8) {
                                LSText("Refine route...", variant: .body.md)
                                Spacer()
                                LSIcon(name: .send, size: .sm, color: .primary)
                            }
                        }
                        .frame(height: 46)
                    }
                ],
                bottomSheet: BottomSheetSpec(detent: .medium) {
                    VStack(alignment: .leading, spacing: 12) {
                        LSText("The Skyline Spine", variant: .heading.lg)
                        HStack(spacing: 8) {
                            LSGlassPanel(variant: .chrome) {
                                LSText("Save", variant: .label.md)
                            }
                            LSGlassPanel(variant: .chrome) {
                                LSText("Ride this", variant: .label.md, color: .primary)
                            }
                        }
                    }
                    .frame(maxWidth: .infinity, alignment: .leading)
                },
                topBar: LSTopBar(
                    onMenuTap: {},
                    onNewTap: {}
                )
            )
        },

        // Dark mode variants
        Story(
            id: "organisms.maplayer.mapOnlyDark",
            tier: .organism,
            component: "MapLayer",
            name: "Map Only · Dark",
            summary: "map-paper resolves to #1B140E. Grid contours use ink-tinted alpha."
        ) { _ in
            LSMapLayer {
                LSMap(
                    mode: .preview,
                    camera: CameraPosition(
                        center: LatLng(lat: 37.7749, lon: -122.4194),
                        zoom: 12.0
                    )
                )
            }
            .preferredColorScheme(.dark)
        },

        Story(
            id: "organisms.maplayer.fullStackDark",
            tier: .organism,
            component: "MapLayer",
            name: "Full Stack · Dark",
            summary: "All seven slots on dark canvas. Deeper scrim, ink-alpha glass borders."
        ) { _ in
            LSMapLayer(
                map: {
                    LSMap(
                        mode: .preview,
                        camera: CameraPosition(
                            center: LatLng(lat: 37.7749, lon: -122.4194),
                            zoom: 12.0
                        )
                    )
                },
                scrim: ScrimSpec(opacity: 0.2),
                topOverlays: [
                    GlassOverlaySlot(id: "navigator") {
                        LSGlassPanel(variant: .chrome) {
                            VStack(alignment: .leading, spacing: 4) {
                                LSText("THE NAVIGATOR", variant: .label.sm, color: .primary)
                                LSText("Best route selected.", variant: .heading.sm)
                            }
                            .frame(maxWidth: .infinity, alignment: .leading)
                        }
                    }
                ],
                bottomOverlays: [
                    GlassOverlaySlot(id: "chat") {
                        LSGlassPanel(variant: .chrome) {
                            HStack(spacing: 8) {
                                LSText("Refine route...", variant: .body.md)
                                Spacer()
                                LSIcon(name: .send, size: .sm, color: .primary)
                            }
                        }
                        .frame(height: 46)
                    }
                ],
                bottomSheet: BottomSheetSpec(detent: .medium) {
                    VStack(alignment: .leading, spacing: 12) {
                        LSText("The Skyline Spine", variant: .heading.lg)
                        HStack(spacing: 8) {
                            LSGlassPanel(variant: .chrome) {
                                LSText("Save", variant: .label.md)
                            }
                            LSGlassPanel(variant: .chrome) {
                                LSText("Ride this", variant: .label.md, color: .primary)
                            }
                        }
                    }
                    .frame(maxWidth: .infinity, alignment: .leading)
                },
                topBar: LSTopBar(
                    onMenuTap: {},
                    onNewTap: {}
                )
            )
            .preferredColorScheme(.dark)
        },
    ]
}

File: android/app/src/debug/java/com/laneshadow/sandbox/stories/organisms/LSMapLayerStory.kt
package com.laneshadow.sandbox.stories.organisms

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.laneshadow.theme.LaneShadowTheme
import com.laneshadow.ui.atoms.CameraPosition
import com.laneshadow.ui.atoms.LatLng
import com.laneshadow.ui.atoms.LSMap
import com.laneshadow.ui.atoms.MapMode
import com.laneshadow.ui.molecules.BottomSheetDetent
import com.laneshadow.ui.molecules.LSBottomSheet
import com.laneshadow.ui.organisms.BottomSheetSpec
import com.laneshadow.ui.organisms.DrawerSpec
import com.laneshadow.ui.organisms.GlassOverlaySlot
import com.laneshadow.ui.organisms.LSMapLayer
import com.laneshadow.ui.organisms.LSTopBar
import com.laneshadow.ui.organisms.SheetDetent
import com.laneshadow.ui.organisms.ScrimSpec
import com.laneshadow.ui.organisms.TopBarTrailing
import com.nativesandbox.model.ComponentTier
import com.nativesandbox.model.Story

object LSMapLayerStory {
    val all: List<Story> = listOf(
        Story(
            id = "organisms.maplayer.map-only",
            tier = ComponentTier.Organism,
            component = "LSMapLayer",
            name = "Map Only",
            summary = "LSMap(mode:.preview) fills the entire phone-screen. No other slots populated.",
            content = { MapOnlyStory() },
        ),
        Story(
            id = "organisms.maplayer.map-topbar",
            tier = ComponentTier.Organism,
            component = "LSMapLayer",
            name = "Map + TopBar",
            summary = "LSTopBar at z-index 5, above everything. Map below. Matches Idle screen pattern.",
            content = { MapTopBarStory() },
        ),
        Story(
            id = "organisms.maplayer.map-top-overlay",
            tier = ComponentTier.Organism,
            component = "LSMapLayer",
            name = "Map + Top Overlay",
            summary = "NavigatorMessage in topOverlays slot. Positioned below TopBar with safe-area padding.",
            content = { MapTopOverlayStory() },
        ),
        Story(
            id = "organisms.maplayer.map-bottom-overlay",
            tier = ComponentTier.Organism,
            component = "LSMapLayer",
            name = "Map + Bottom Overlay",
            summary = "ChatInput in bottomOverlays. Anchored above bottom safe area at z-index 2.",
            content = { MapBottomOverlayStory() },
        ),
        Story(
            id = "organisms.maplayer.map-scrim-drawer",
            tier = ComponentTier.Organism,
            component = "LSMapLayer",
            name = "Map + Scrim + Drawer",
            summary = "LSScrim z-1, SessionsDrawer at z-4 above scrim. Matches SCR-05 Sessions pattern.",
            content = { MapScrimDrawerStory() },
        ),
        Story(
            id = "organisms.maplayer.map-sheet",
            tier = ComponentTier.Organism,
            component = "LSMapLayer",
            name = "Map + Sheet",
            summary = "RouteSheet at z-index 3, anchored bottom. TopBar still visible above sheet at z-5.",
            content = { MapSheetStory() },
        ),
        Story(
            id = "organisms.maplayer.full-stack",
            tier = ComponentTier.Organism,
            component = "LSMapLayer",
            name = "Full Stack",
            summary = "All slots populated: map + scrim + top/bottom overlays + sheet + top bar. Maximum composition story.",
            content = { FullStackStory() },
        ),
    )
}

@Composable
private fun MapOnlyStory() {
    LaneShadowTheme {
        LSMapLayer(
            map = {
                LSMap(
                    mode = MapMode.Preview,
                    camera = CameraPosition(
                        center = LatLng(37.7749, -122.4194),
                        zoom = 12.0,
                    ),
                )
            },
        )
    }
}

@Composable
private fun MapTopBarStory() {
    LaneShadowTheme {
        LSMapLayer(
            map = {
                LSMap(
                    mode = MapMode.Preview,
                    camera = CameraPosition(
                        center = LatLng(37.7749, -122.4194),
                        zoom = 12.0,
                    ),
                )
            },
            topBar = {
                LSTopBar(
                    onMenuTap = {},
                    onNewTap = {},
                )
            },
        )
    }
}

@Composable
private fun MapTopOverlayStory() {
    LaneShadowTheme {
        LSMapLayer(
            map = {
                LSMap(
                    mode = MapMode.Preview,
                    camera = CameraPosition(
                        center = LatLng(37.7749, -122.4194),
                        zoom = 12.0,
                    ),
                )
            },
            topBar = {
                LSTopBar(
                    onMenuTap = {},
                    onNewTap = {},
                )
            },
            topOverlays = listOf(
                GlassOverlaySlot("navigator-message") {
                    Column(
                        modifier = Modifier
                            .padding(com.laneshadow.theme.LocalLaneShadowTheme.current.space.md),
                        verticalArrangement = Arrangement.spacedBy(com.laneshadow.theme.LocalLaneShadowTheme.current.space.sm),
                    ) {
                        com.laneshadow.ui.atoms.LSText(
                            text = "THE NAVIGATOR",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Ui.Label.Sm,
                            color = com.laneshadow.ui.atoms.ContentColor.Primary,
                        )
                        com.laneshadow.ui.atoms.LSText(
                            text = "Here are three routes south through the coastal hills.",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Opinion.Md,
                            color = com.laneshadow.ui.atoms.ContentColor.Primary,
                        )
                    }
                }
            ),
        )
    }
}

@Composable
private fun MapBottomOverlayStory() {
    LaneShadowTheme {
        LSMapLayer(
            map = {
                LSMap(
                    mode = MapMode.Preview,
                    camera = CameraPosition(
                        center = LatLng(37.7749, -122.4194),
                        zoom = 12.0,
                    ),
                )
            },
            topBar = {
                LSTopBar(
                    onMenuTap = {},
                    onNewTap = {},
                )
            },
            bottomOverlays = listOf(
                GlassOverlaySlot("chat-input") {
                    com.laneshadow.ui.molecules.LSFormField(
                        label = "Chat",
                        value = "Where do you want to ride?",
                        onValueChange = {},
                        placeholder = "Where do you want to ride?",
                    )
                }
            ),
        )
    }
}

@Composable
private fun MapScrimDrawerStory() {
    LaneShadowTheme {
        LSMapLayer(
            map = {
                LSMap(
                    mode = MapMode.Preview,
                    camera = CameraPosition(
                        center = LatLng(37.7749, -122.4194),
                        zoom = 12.0,
                    ),
                )
            },
            scrim = ScrimSpec(opacity = 0.35f),
            leadingDrawer = DrawerSpec(
                content = {
                    Column(
                        modifier = Modifier
                            .padding(com.laneshadow.theme.LocalLaneShadowTheme.current.space.md),
                        verticalArrangement = Arrangement.spacedBy(com.laneshadow.theme.LocalLaneShadowTheme.current.space.sm),
                    ) {
                        com.laneshadow.ui.atoms.LSText(
                            text = "Rides",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Ui.Title.Lg,
                            color = com.laneshadow.ui.atoms.ContentColor.Primary,
                        )
                        com.laneshadow.ui.atoms.LSText(
                            text = "Santa Cruz Loop",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Ui.Body.Md,
                            color = com.laneshadow.ui.atoms.ContentColor.Primary,
                        )
                    }
                },
                onDismiss = {},
            ),
        )
    }
}

@Composable
private fun MapSheetStory() {
    LaneShadowTheme {
        LSMapLayer(
            map = {
                LSMap(
                    mode = MapMode.Preview,
                    camera = CameraPosition(
                        center = LatLng(37.7749, -122.4194),
                        zoom = 12.0,
                    ),
                )
            },
            topBar = {
                LSTopBar(
                    onMenuTap = {},
                )
            },
            bottomSheet = BottomSheetSpec(
                content = {
                    Column(
                        modifier = Modifier
                            .padding(com.laneshadow.theme.LocalLaneShadowTheme.current.space.md),
                        verticalArrangement = Arrangement.spacedBy(com.laneshadow.theme.LocalLaneShadowTheme.current.space.sm),
                    ) {
                        com.laneshadow.ui.atoms.LSText(
                            text = "The Skyline Spine",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Opinion.Lg,
                            color = com.laneshadow.ui.atoms.ContentColor.Primary,
                        )
                        com.laneshadow.ui.atoms.LSText(
                            text = "via Kings Mountain Rd · 47 mi",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Ui.Body.Sm,
                            color = com.laneshadow.ui.atoms.ContentColor.Secondary,
                        )
                    }
                },
                detent = SheetDetent.Medium,
                onDismiss = {},
            ),
        )
    }
}

@Composable
private fun FullStackStory() {
    LaneShadowTheme {
        LSMapLayer(
            map = {
                LSMap(
                    mode = MapMode.Preview,
                    camera = CameraPosition(
                        center = LatLng(37.7749, -122.4194),
                        zoom = 12.0,
                    ),
                )
            },
            scrim = ScrimSpec(opacity = 0.2f),
            topBar = {
                LSTopBar(
                    onMenuTap = {},
                    onNewTap = {},
                )
            },
            topOverlays = listOf(
                GlassOverlaySlot("navigator-message") {
                    Column(
                        modifier = Modifier
                            .padding(com.laneshadow.theme.LocalLaneShadowTheme.current.space.md),
                        verticalArrangement = Arrangement.spacedBy(com.laneshadow.theme.LocalLaneShadowTheme.current.space.sm),
                    ) {
                        com.laneshadow.ui.atoms.LSText(
                            text = "THE NAVIGATOR",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Ui.Label.Sm,
                            color = com.laneshadow.ui.atoms.ContentColor.Primary,
                        )
                        com.laneshadow.ui.atoms.LSText(
                            text = "Best route selected.",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Opinion.Md,
                            color = com.laneshadow.ui.atoms.ContentColor.Primary,
                        )
                    }
                }
            ),
            bottomOverlays = listOf(
                GlassOverlaySlot("chat-input") {
                    com.laneshadow.ui.molecules.LSFormField(
                        label = "Chat",
                        value = "Refine route...",
                        onValueChange = {},
                        placeholder = "Refine route...",
                    )
                }
            ),
            bottomSheet = BottomSheetSpec(
                content = {
                    Column(
                        modifier = Modifier
                            .padding(com.laneshadow.theme.LocalLaneShadowTheme.current.space.md),
                        verticalArrangement = Arrangement.spacedBy(com.laneshadow.theme.LocalLaneShadowTheme.current.space.sm),
                    ) {
                        com.laneshadow.ui.atoms.LSText(
                            text = "The Skyline Spine",
                            variant = com.laneshadow.ui.atoms.TypographyVariant.Opinion.Lg,
                            color = com.laneshadow.ui.atoms.ContentColor.Primary,
                        )
                    }
                },
                detent = SheetDetent.Medium,
                onDismiss = {},
            ),
        )
    }
}
